timer/timer.go

```diff
@@ -0,0 +1,144 @@
+package timer
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// Timer definition
+type (
+	Timer struct {
+		ID             uuid.UUID
+		Domain         string
+		ExecutionCount uint32
+		Meta           Meta
+		Task           Task
+		Schedule       Schedule
+	}
+
+	// Task represents a timer task.
+	Task interface {
```

Comment:
This is weird to have an exported interface with its only method be non-exported.

Reply:
This makes it so others can type switch on it but not implement it.